cmd/skaffold/app/cmd/cmd.go

```diff
@@ -75,26 +82,29 @@ func SetUpLogs(out io.Writer, level string) error {
 	return nil
 }
 
-var defaultConfigForMode = map[bool]*config.SkaffoldConfig{
-	true:  config.DefaultDevSkaffoldConfig,
-	false: config.DefaultRunSkaffoldConfig,
-}
-
 func runSkaffold(out io.Writer, dev bool, filename string) error {
-	buf, err := readConfiguration(filename)
+	buf, err := util.ReadConfiguration(filename)
 	if err != nil {
 		return errors.Wrap(err, "read skaffold config")
 	}
 
-	defaultConfig := defaultConfigForMode[dev]
-	cfg, err := config.Parse(buf, defaultConfig)
+	apiVersion := &config.ApiVersion{}
+	if err := yaml.Unmarshal(buf, apiVersion); err != nil {
```

Comment:
I'm a little worried about fix and dev/run going through the same codepath. Can we hardcode the latest version here?

Would it be possible to keep this machinery the same and only use the conversion code in fix?

Reply:
The only codepath they really share is the config parsing, fix just tries to parse the config first to see if it's already the latest version. `config.Parse()` doesn't really do anything fancy except iterate through config versions and attempt to parse into each one, the transforms all happen separately and aren't touched by dev/run.

I'm not huge on hardcoding the latest version here, it's already basically hardcoded through the type definition in config/config.go.